File: src/main/java/org/radarbase/management/service/dto/AttributeMapDTO.java
package org.radarbase.management.service.dto;

import java.util.Objects;

/**
 * Created by nivethika on 30-8-17.
 */
public class AttributeMapDTO {
    private String key;

    private String value;

    public AttributeMapDTO() {
        this(null, null);
    }

    public AttributeMapDTO(String key, String value) {
        this.key = key;
        this.value = value;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        AttributeMapDTO attributeMapDto = (AttributeMapDTO) o;

        return Objects.equals(key, attributeMapDto.key)
                && Objects.equals(value, attributeMapDto.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(key, value);
    }

    @Override
    public String toString() {
        return "AttributeMapDTO{"
                + " key='" + key + "'"
                + ", value='" + value + "'"
                + '}';
    }
}

File: src/main/java/org/radarbase/management/service/dto/AttributeMapDTO.kt
package org.radarbase.management.service.dto

import java.util.*

/**
 * Created by nivethika on 30-8-17.
 */
class AttributeMapDTO
    @JvmOverloads
    constructor(
        var key: String? = null,
        var value: String? = null,
    ) {
        override fun equals(other: Any?): Boolean {
            if (this === other) {
                return true
            }
            if (other == null || javaClass != other.javaClass) {
                return false
            }
            val attributeMapDto = other as AttributeMapDTO
            return key == attributeMapDto.key && value == attributeMapDto.value
        }

        override fun hashCode(): Int = Objects.hash(key, value)

        override fun toString(): String =
            (
                "AttributeMapDTO{" +
                    " key='" + key + "'" +
                    ", value='" + value + "'" +
                    '}'
                )
    }
